### PR TITLE
chore: add a trigger type argument to DropDown onTrigger

### PR DESCRIPTION
**Pull request by reedham1**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**
- Added string parameter to the onTrigger function to specify the type of trigger (so parent components can build custom logic around dropdown)
- Unit tests

### CODE REVIEW

File: src/DropDown/DropDown.component.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ interface DropDownProps extends React.HTMLAttributes<HTMLDivElement> {
     border?: string;
     borderRadius?: string;
     debug?: boolean;
-    onTriggered?: (state: boolean) => void;
+    onTriggered?: (state: boolean, trigger?: string) => void;
```
Comment (sharkparty) [inline src/DropDown/DropDown.component.tsx:27]: This is something we can unit test, and should. What's the end-goal of changing this util method?

Comment (reedham1) [reply src/DropDown/DropDown.component.tsx:27]: We have a button in a navbar that you click/hover to trigger the dropdown, and we want the button to respond to the dropdown (color change, chevron rotate, etc). Wanting to use `onTriggered` to track the state of the dropdown, but it is not useful when using the 'both' value for the `trigger` prop.

Happy to write unit tests if you're ok with the change

Comment (reedham1) [reply src/DropDown/DropDown.component.tsx:27]: If the button knows which trigger type is firing `onTriggered`, we can write a bit of logic to respond appropriately

**Review by sharkparty — APPROVED** [review]
LGTM

**Pull request merged**